src/detect-flowbits.c

```diff
@@ -1364,6 +1372,35 @@ static int FlowBitsTestSig11(void)
     PASS;
 }
 
+/**
+ * \test FlowBitsTestSig12 is a test to check random arguments to
+ *  flowbits keyword are rejected
+ *  See https://redmine.openinfosecfoundation.org/issues/5154
+ *  \retval 1 on succces
+ *  \retval 0 on failure
+ */
+
+static int FlowBitsTestSig12(void)
+{
+    Signature *s = NULL;
+    DetectEngineCtx *de_ctx = NULL;
+
+    de_ctx = DetectEngineCtxInit();
+    FAIL_IF_NULL(de_ctx);
+
+    de_ctx->flags |= DE_QUIET;
+
+    s = de_ctx->sig_list =
+            SigInit(de_ctx, "alert http any any -> any any (msg:\"flowbits with noalert option\"; "
```

Comment:
prefer `DetectEngineAppendSig` for new code. That way you don't have to do the `de_ctx->sig_list =` assign